frontend/src/component/common/Combo/Combo.tsx

```diff
@@ -0,0 +1,150 @@
+import {
+    HTMLAttributes,
+    forwardRef,
+    useCallback,
+    useRef,
+    useState,
+} from 'react';
+import { Autocomplete, AutocompleteProps, styled } from '@mui/material';
+import { GroupedVirtuoso, GroupedVirtuosoHandle } from 'react-virtuoso';
+
+const VIRTUALIZED_THRESHOLD = 100;
+
+const StyledDiv = styled('div')(({ theme }) => ({
+    'div[data-item-group-index]': {
+        '.MuiAutocomplete-option': {
+            paddingLeft: theme.spacing(3),
+        },
+    },
+}));
+
+interface IComboProps<
+    T,
+    Multiple extends boolean | undefined,
+    DisableClearable extends boolean | undefined,
+    FreeSolo extends boolean | undefined
+> extends AutocompleteProps<T, Multiple, DisableClearable, FreeSolo> {
+    virtualized?: 'auto' | 'always' | 'never' | undefined;
+}
+
+export const Combo = <
+    T,
+    Multiple extends boolean | undefined = false,
+    DisableClearable extends boolean | undefined = false,
+    FreeSolo extends boolean | undefined = false
+>({
+    virtualized = 'auto',
+    ...props
+}: IComboProps<T, Multiple, DisableClearable, FreeSolo>) => {
+    if (
+        virtualized === 'never' ||
+        (virtualized === 'auto' && props.options.length < VIRTUALIZED_THRESHOLD)
+    )
+        return <Autocomplete {...props} />;
```

Comment:
Use plain old `Autocomplete` if we don't match the necessary conditions for virtualization.